src/detect.h

```diff
@@ -1291,7 +1291,7 @@ typedef struct SigTableElmt_ {
         uint8_t flags, File *, const Signature *, const SigMatchCtx *);
 
     /** InspectionBuffer transformation callback */
-    void (*Transform)(InspectionBuffer *, void *context);
+    void (*Transform)(DetectEngineThreadCtx *, InspectionBuffer *, void *context);
```

Comment:
This requires to bump up `SC_PLUGIN_API_VERSION`

Reply:
> This requires to bump up `SC_PLUGIN_API_VERSION`

Why? Its currently at 8, and should stay that way until 8 is released.

Reply:
Our compatibility promises are for a major version.  So 8 is fine, as 8 is not released there is no promise that its stable until 8 is released.

Reply:
We'll need work on the versioning, so I think its out of scope for this PR.

For example, if we ever did an 8.1 release, we wouldn't consider that compatible with 8.0.  So we'd want something like:
- 800000 # 8.0
- 800100 # 8.1
Patch releases should not be considered as they should not break the API. I'd still not consider pre-release git snapshot versions either.

Reply:
I think you forget to take into account the people using `git master` (like after 8 release before 9)

> We'll need to work on the versioning

indeed

> so I think its out of scope for this PR.

But a change in `SC_PLUGIN_API_VERSION` is required by this PR

> As the library matures we'll end up in a position where any change to any public function signature will require an update to this, which could be a lot of commits during a major development cycle.

I think we are there

Reply:
In git master we make no promises until it is released as a final 8.0.0 release. Ppl relying on a dev branch will have to accept we can still break things.

Reply:
I understand that.

Even bumping `SC_PLUGIN_API_VERSION`, this PR still "breaks things", but it will give a better error message instead of a segfault.

This is to get the good habit for post 8 release

Reply:
I understand that.

Even bumping `SC_PLUGIN_API_VERSION`, this PR still "breaks things", but it will give a better error message instead of a segfault.

This is to get the good habit for post 8 release